yt/yt/server/node/exec_node/job.cpp

```diff
@@ -2563,13 +2563,21 @@ TJobProxyInternalConfigPtr TJob::CreateConfig()
         }
     }
 
+    auto calculateShardingKey = [&] (size_t shardingKeyLength) {
+        auto randomPart = GetId().Underlying().Parts64[1];
```

Comment:
I think we should introduce additional function like `RandomPartFromAllocationId(TAllocationId allocationId)` and use it here (since it is not obvious at this point that `Parts64[1]` is a random one)